compiler/noirc_evaluator/src/acir/tests/mod.rs

```diff
@@ -924,3 +925,208 @@ fn do_not_overflow_with_constant_constrain_neq() {
+        }}"
+            ),
+            false,
+        )
+    }
+}
+
+/// Execute a simple operation for each operators
+/// The operation is executed from SSA IR using ACVM after acir-gen
+/// and also directly on the SSA IR using the SSA interpreter.
+/// The results are compared to ensure that both executions yield the same result.
+fn test_operators(
+    // The list of operators to test
+    operators: &[&str],
+    // the type of the input values
+    typ: &str,
+    // the values of the inputs
+    inputs: &[FieldElement],
+) {
+    let main = get_main_src(typ);
+    let num_type = match typ.chars().next().unwrap() {
+        'F' => NumericType::NativeField,
+        'i' => NumericType::Signed { bit_size: typ[1..].parse().unwrap() },
+        'u' => NumericType::Unsigned { bit_size: typ[1..].parse().unwrap() },
+        _ => unreachable!("invalid numeric type"),
+    };
+    let inputs_int = Value::array_from_iter(inputs.iter().cloned(), num_type).unwrap();
+    let inputs =
+        inputs.iter().enumerate().map(|(i, f)| (Witness(i as u32), *f)).collect::<BTreeMap<_, _>>();
+    let len = inputs.len() as u32;
+    let initial_witness = WitnessMap::from(inputs);
+
+    for op in operators {
+        let (src, with_output) = generate_test_instruction_from_operator(op);
+        let output = if with_output { Some(Witness(len)) } else { None };
+        let ssa = Ssa::from_str(&(main.to_owned() + &src)).unwrap();
+        // ssa execution
+        let ssa_interpreter_result = ssa.interpret(vec![inputs_int.clone()]);
+        // acir execution
+        let acir_execution_result = execute_ssa(ssa, initial_witness.clone(), output.as_ref());
+
+        match (ssa_interpreter_result, acir_execution_result) {
+            // Both execution failed, so it is the same behavior, as expected.
+            (Err(_), (ACVMStatus::Failure(_), _)) => (),
+            // Both execution succeeded and output the same value
+            (Ok(ssa_inner_result), (ACVMStatus::Solved, acvm_result)) => {
+                let ssa_result = if let Some(result) = ssa_inner_result.first() {
+                    result.as_numeric().map(|v| v.convert_to_field())
+                } else {
+                    None
+                };
+                assert_eq!(ssa_result, acvm_result);
+            }
+            _ => panic!("ssa and acvm execution should have the same result"),
+        }
+    }
+}
+
+#[test]
+fn test_binary_on_field() {
+    // Test the following Binary operation on Fields
+    let operators = [
+        "add",
+        "sub",
+        "mul",
+        "div",
+        "eq",
+        // Bitwise operations are not allowed on field elements
+        // SSA interpreter will emit an error but not ACVM
+        // "and",
+        // "xor",
+        "unchecked_add",
+        "unchecked_sub",
+        "unchecked_mul",
+        "range_check 32",
+        "truncate 32 254",
+    ];
+    let inputs = [FieldElement::from(1_usize), FieldElement::from(2_usize)];
```

Comment:
We're only testing a single input here which means we're not going to cover a lot of interesting inputs. Can you look at tying this into proptest so we can compile the circuits once and then throw a lot of inputs into it?